lib/utilities/fetch.ts

```diff
@@ -0,0 +1,17 @@
+import fetch from 'isomorphic-unfetch'
+
+export default async function fetchRetry (url: string, requestConfig: RequestInit, retries = 1): Promise<Response> {
+  const response = await fetch(url, requestConfig)
+  if (response?.ok) {
+    return response
+  }
+
+  if (retries > 0) {
+    console.log(`Fetch failed with status ${response?.status}, trying again... (attempts remaining ${retries - 1})`)
```

Comment:
These console logs may be here for draft purposes, but if not, you could perhaps use `console.error` instead.